From 05b3a23979ed4686ae65e649cfed762c45986660 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 18 Aug 2022 06:17:22 +0300
Subject: [PATCH] [SW-99461] gaudi3: add boot info skip api

As per the updated efuse sheet for gaudi3, efuse bit 14 is to be
used to skip boot info usage in ROM.

This patch adds the api to check for the efuse bit status.

Change-Id: Ie66ccb7595c36081c079fd9c4dbbd7212f8d0dee
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 05b7839..95723a9 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1122,3 +1122,11 @@
 
 	return 0;
 }
+
+int arcboot_rom_efuse_boot_info_skip(void)
+{
+	/* check efuse for boot info skip */
+	uint32_t config_fuse = arcboot_rom_global.sec_data.efuse_sys_cfg;
+
+	return !!(config_fuse & EFUSE_SYSCFG_BOOT_INFO_SKIP);
+}
diff --git a/arch/gaudi3/arc/dev_common.h b/arch/gaudi3/arc/dev_common.h
index 1cbf795..13419cf 100644
--- a/arch/gaudi3/arc/dev_common.h
+++ b/arch/gaudi3/arc/dev_common.h
@@ -34,3 +34,4 @@
 enum dma_dest_type arcboot_rom_get_dma_dest_type(uint64_t addr);
 void arcboot_rom_halt(void);
 bool arcboot_is_valid_start_addr(struct boot_desc *desc);
+int arcboot_rom_efuse_boot_info_skip(void);
diff --git a/arch/gaudi3/arc/dev_efuse.h b/arch/gaudi3/arc/dev_efuse.h
index 6792139..cf4c058 100644
--- a/arch/gaudi3/arc/dev_efuse.h
+++ b/arch/gaudi3/arc/dev_efuse.h
@@ -21,6 +21,7 @@
 #define EFUSE_SYSCFG_RECOVERY_CODE_LOCATION_MASK	BIT(7)
 #define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK		BIT(11)
 #define EFUSE_SYSCFG_EFUSE_LOCATION_MASK		BIT(13)
+#define EFUSE_SYSCFG_BOOT_INFO_SKIP			BIT(14)
 #define EFUSE_SYSCFG_SECURITY_ENABLE_MASK		BIT(15)
 #define EFUSE_SYSCFG_RR_MASK				BIT(16)
 #define EFUSE_SYSCFG_ENABLE_TPM_MASK			BIT(17)
